pkg/client/openshift/buildconfig.go

```diff
@@ -0,0 +1,137 @@
+package openshift
+
+import (
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/meta"
+	buildv1 "github.com/openshift/api/build/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// BuildState describes the state of the build
+type BuildState struct {
+	ImageExists  bool
+	BuildRunning bool
+}
+
+// BuildConfigInterface exposes OpenShift BuildConfig operations
+type BuildConfigInterface interface {
+	// EnsureImageBuild checks for the corresponding image for the build. If there's no image, verifies if the build still running.
+	// Returns a BuildState structure describing it's results. Label selector is used to query for the right bc
+	EnsureImageBuild(bc *buildv1.BuildConfig, labelSelector string) (*BuildState, error)
```

Comment:
Why is the return type, build state, a pointer? This question also potentially applies to buildconfig and others.

Reply:
Yeah, for those types doesn't make sense returning a pointer.